sdcm/stress_thread.py

```diff
@@ -370,3 +370,27 @@ def db_node_to_query(self, loader):
             LOGGER.debug("Selected '%s' to query for local nodes", node_to_query)
             return node_to_query.cql_ip_address
         return self.node_list[0].cql_ip_address
+
+
+duration_pattern = re.compile(r'(?P<hours>[\d]*)h|(?P<minutes>[\d]*)m|(?P<seconds>[\d]*)s')
+stress_cmd_get_duration_pattern = re.compile(r' [-]{0,2}duration[\s=]+([\d]+[hms]+)')
+stress_cmd_get_warmup_pattern = re.compile(r' [-]{0,2}warmup[\s=]+([\d]+[hms]+)')
+
+
+def time_period_str_to_seconds(time_str: str) -> int:
+    """Transforms duration string into seconds int. e.g. 1h -> 3600, 1h22m->4920 or 10m->600"""
+    return sum([int(g[0] or 0) * 3600 + int(g[1] or 0) * 60 + int(g[2] or 0) for g in duration_pattern.findall(time_str)])
```

Comment:
what is `g`?
couldn't it have a better name?

Reply:
possibly it could have better name,`g` is just for brevity (it is from `group`). Function docs should be enough to understand how it works.